org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandler.java

```diff
@@ -80,6 +88,37 @@ public SignatureHelp signatureHelp(TextDocumentPositionParams position, IProgres
 		return help;
 	}
 
+	private boolean isValid(ICompilationUnit unit, int offset, IProgressMonitor monitor) {
+		CompilationUnit ast = CoreASTProvider.getInstance().getAST(unit, CoreASTProvider.WAIT_YES, monitor);
+		boolean[] valid = { false };
+		ast.accept(new ASTVisitor() {
+
+			/* (non-Javadoc)
+			 * @see org.eclipse.jdt.core.dom.ASTVisitor#visit(org.eclipse.jdt.core.dom.MethodRef)
+			 */
+			@Override
+			public boolean visit(MethodRef node) {
+				if (node.getStartPosition() <= offset && (node.getStartPosition() + node.getLength()) >= offset) {
+					valid[0] = true;
+				}
+				return super.visit(node);
+			}
+
+			/* (non-Javadoc)
+			 * @see org.eclipse.jdt.core.dom.ASTVisitor#endVisit(org.eclipse.jdt.core.dom.MethodInvocation)
+			 */
+			@Override
+			public void endVisit(MethodInvocation node) {
+				if (node.getStartPosition() <= offset && (node.getStartPosition() + node.getLength()) >= offset) {
+					valid[0] = true;

```

Comment:
will this method be called after visit(MethodRef) or they're really for 2 different node types?

Reply:
They are different node types.
```
org.eclipse.jdt.core.dom.MethodInvocation
Method invocation expression AST node type.
```

```
org.eclipse.jdt.core.dom.MethodRef

AST node for a method or constructor reference within a doc comment (Javadoc). The principal uses of these are in "@see" and "@link" tag elements, for references to method and constructor members.
```
I have added a new test.

Reply:
we don't need signature help in javadoc, or do we?

Reply:
It could be useful.